EssentialCSharp.Web/Extensions/StringExtensions.cs

```diff
@@ -65,4 +65,17 @@ public static string KeyToHeading(this string str)
     {
         return System.Globalization.CultureInfo.CurrentCulture.TextInfo.ToTitleCase(str.Trim().ToLowerInvariant().Replace('-', ' '));
     }
+
+    public static SiteMapping? Find(this string key, IList<SiteMapping> siteMappings)
```

Comment:
Having this method on `string` feels a little off. What about inverting the order of the parameters and making the `this` parameter `IList<SiteMapping> siteMappings`? Perhaps renaming it to "FindByKey" or similar?